LeadKit/LeadKit/Extensions/UIImage/UIImage+Alpha.swift

```diff
@@ -0,0 +1,123 @@
+                           CGRect(x: 0, y: 0, width: width, height: height),
+                           imageRef)
+        guard let cgImage = CGBitmapContextCreateImage(offscreenContext) else {
+            return nil
+        }
+        let imageWithAlpha = UIImage(CGImage: cgImage)
+        return imageWithAlpha
+    }
+
+    /**
+     returns a copy of the image with a transparent border of the given size added around its edges.
+     i.e. For rotating an image without getting jagged edges.
+
+     - parameter padding: The padding amount.
+
+     - returns: A new image.
+     */
+    public func applyPadding(padding: CGFloat) -> UIImage? {
+        // If the image does not have an alpha layer, add one
+        guard let image = applyAlpha() else {
+            return nil
+        }
+        let rect = CGRect(x: 0, y: 0, width: size.width + padding * 2, height: size.height + padding * 2)
+
+        // Build a context that's the same dimensions as the new size
+        let colorSpace = CGImageGetColorSpace(CGImage)
+        let bitmapInfo = CGImageGetBitmapInfo(CGImage)
+        let bitsPerComponent = CGImageGetBitsPerComponent(CGImage)
+        let context = CGBitmapContextCreate(nil,
+                                            Int(rect.size.width),
+                                            Int(rect.size.height),
+                                            bitsPerComponent, 0, colorSpace,
+                                            bitmapInfo.rawValue)
+
+        // Draw the image in the center of the context, leaving a gap around the edges
+        let imageLocation = CGRect(x: padding, y: padding, width: image.size.width, height: image.size.height)
+        CGContextDrawImage(context, imageLocation, CGImage)
+
+        // Create a mask to make the border transparent, and combine it with the image
+        let imageWithPadding = imageRefWithPadding(padding, size: rect.size)
+        guard let cgImage = CGImageCreateWithMask(CGBitmapContextCreateImage(context), imageWithPadding) else {
+            return nil
+        }
+        let transparentImage = UIImage(CGImage: cgImage)
+        return transparentImage
+    }
+
+    /**
+     creates a mask that makes the outer edges transparent and everything else opaque. 
+     The size must include the entire mask (opaque part + transparent border).
+
+     - parameter padding: The padding amount.
+     - parameter size: The size of the image.
+
+     - returns: A Core Graphics Image Ref
+     */
+    private func imageRefWithPadding(padding: CGFloat,
+                                     size: CGSize) -> CGImageRef? {
+        // Build a context that's the same dimensions as the new size
+        let colorSpace = CGColorSpaceCreateDeviceGray()
+        let bitmapInfo = CGBitmapInfo(rawValue: CGBitmapInfo.ByteOrderDefault.rawValue | CGImageAlphaInfo.None.rawValue)
+        let context = CGBitmapContextCreate(nil, Int(size.width), Int(size.height), 8, 0, colorSpace, bitmapInfo.rawValue)
```

Comment:
why not
```swift
init?(data: UnsafeMutableRawPointer?, 
width: Int, 
height: Int, 
bitsPerComponent: Int, 
bytesPerRow: Int, 
space: CGColorSpace, 
bitmapInfo: UInt32)
```
?
https://developer.apple.com/reference/coregraphics/cgcontext/1455939-init